packages/venia-concept/src/components/ProductImageCarousel/thumbnail.js

```diff
@@ -10,19 +11,32 @@ class Thumbnail extends Component {
     static propTypes = {
         classes: PropTypes.shape({
             root: PropTypes.string
-        })
+        }),
+        item: imageItemPropType,
+        activeItemSrc: PropTypes.string,
+        onClickHandler: PropTypes.func,
+        handlerParams: PropTypes.string
     };
 
     render() {
-        const { classes, item } = this.props;
-        const src = item.file
-            ? makeProductMediaPath(item.file)
-            : transparentPlaceholder;
+        const {
+            classes,
+            item: { file, label },
+            onClickHandler,
+            activeItemSrc,
+            handlerParams
+        } = this.props;
+        const src = file ? makeProductMediaPath(file) : transparentPlaceholder;
 
         return (
-            <div className={classes.root}>
-                <img className={classes.image} src={src} alt={item.label} />
-            </div>
+            <button
+                onClick={() => onClickHandler(handlerParams)}
```

Comment:
You can add class method, that will take this handler and item from props and pass data to handler as a parameter

Reply:
Yep, It will elevate performance a little